src/ert/scheduler/job.py

```diff
@@ -63,41 +66,58 @@ def driver(self) -> Driver:
         return self._scheduler.driver
 
     async def __call__(
-        self, start: asyncio.Event, sem: asyncio.BoundedSemaphore
+        self, start: asyncio.Event, sem: asyncio.BoundedSemaphore, max_submit: int = 2
     ) -> None:
         await start.wait()
-        await sem.acquire()
-
-        try:
-            await self._send(State.SUBMITTING)
-            await self.driver.submit(
-                self.real.iens, self.real.job_script, cwd=self.real.run_arg.runpath
-            )
-
-            await self._send(State.STARTING)
-            await self.started.wait()
-
-            await self._send(State.RUNNING)
-            while not self.returncode.done():
-                await asyncio.sleep(0.01)
-            returncode = await self.returncode
-            if (
-                returncode == 0
-                and forward_model_ok(self.real.run_arg).status
-                == LoadStatus.LOAD_SUCCESSFUL
-            ):
-                await self._send(State.COMPLETED)
-            else:
-                await self._send(State.FAILED)
-
-        except asyncio.CancelledError:
-            await self._send(State.ABORTING)
-            await self.driver.kill(self.iens)
-
-            await self.aborted.wait()
-            await self._send(State.ABORTED)
-        finally:
-            sem.release()
+        retries = 0
+        retry: bool = True
+        while retry:
```

Comment:
Maybe separate the retry functionality into its own function? This is beginning to be a mess.

Eg:
```py
async def __call__(...):
    for _ in range(max_submit):
        if await self.actually_do_the_thing(...):
            # SUCCESS!
    else:
        # FAILURE (No more retries)
```

Also, using a normal loop will probably avoid having to do these `retries`, `retry` variables and manually checking whether you should escape.

Reply:
> Maybe separate the retry functionality into its own function? This is beginning to be a mess.

Agree, I saw it coming, will give it some tries to split it in 2.

Reply:
Now it's split.